tree.go

```diff
@@ -10,13 +11,17 @@ import (
 )
 
 func addMiddleware(t mux.Tree, method, path string, mid middleware.Middleware) {
+	fmt.Printf("tree: %v\n", t.PrettyPrint())
 	type recFunc func(recFunc, mux.Node, middleware.Middleware)
 
 	c := func(c recFunc, n mux.Node, m middleware.Middleware) {
+		fmt.Printf("route handler: %v\n", n.Route())
```

Comment:
I understand that this is for the debug purpose, but once you decide to finalise your pr
you have to remove all the prints.

Once you believe you came up with valid solution, mark your pull request with `ready to merge` label. This way I will know you want it to be merged.